src/test/java/core/basesyntax/service/operations/PurchaseOperationHandlerTest.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax.service.operations;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dto.FruitTransactionDto;
+import core.basesyntax.exeptions.NegativeNumberExeption;
+import core.basesyntax.exeptions.UnsupportedOperationExeption;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.storage.Storage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PurchaseOperationHandlerTest {
+    private PurchaseOperationHandler purchaseOperationHandler;
+
+    @BeforeEach
+    public void setUp() {
+        purchaseOperationHandler = new PurchaseOperationHandler();
+    }
+
+    @Test
+    public void testApply_ValidTransaction_Ok() {
+        Storage.fruits.put(new Fruit("apple"), 10);
+        FruitTransactionDto dto = new FruitTransactionDto("p", "apple", 5);
+        purchaseOperationHandler.apply(dto);
+
+        assertEquals(5, Storage.fruits.get(new Fruit("apple")));
+    }
```

Comment:
Here you are creating this ^new Fruit("apple")^ at least a few times, so let's create this fruit as a class level field and pass it into the parameters. (Also, you use this "apple" string in every test, so like I said, make the repeated strings and numbers constant)

Reply:
Magic numbers done.
Model Fruit its a record.